concent_api/core/views.py

```diff
@@ -342,33 +432,47 @@ def handle_receive_force_get_task_result_upload_for_provider(request, decoded_me
         force_get_task_result_upload.TYPE,
         decoded_message.report_computed_task.task_to_compute.compute_task_def['task_id'],
         force_get_task_result_upload.serialize(),
-        status    = ReceiveStatus,
-        delivered = True,
+        provider_public_key  = client_public_key,
+        requestor_public_key = previous_message_from_database.message.auth.requestor_public_key_bytes,
+        status               = ReceiveStatus,
+        delivered            = True,
     )
     force_get_task_result_upload.sig = None
     return force_get_task_result_upload
 
 
-def handle_receive_force_get_task_result_failed(decoded_message: message.concents.ForceGetTaskResultUpload) -> message.concents.ForceGetTaskResultUpload:
+def handle_receive_force_get_task_result_failed(
+    request,
+    decoded_message:                message.concents.ForceGetTaskResultUpload,
+    previous_message_from_database: Message
+) -> message.concents.ForceGetTaskResultUpload:
     assert decoded_message.TYPE in message.registered_message_types
 
+    client_public_key = decode_client_public_key(request)
     force_get_task_result_failed = message.concents.ForceGetTaskResultFailed()
     force_get_task_result_failed.task_to_compute = decoded_message.force_get_task_result.report_computed_task.task_to_compute
     store_message_and_message_status(
         force_get_task_result_failed.TYPE,
         force_get_task_result_failed.task_to_compute.compute_task_def['task_id'],
         force_get_task_result_failed.serialize(),
-        status    = ReceiveStatus,
-        delivered = True,
+        provider_public_key  = previous_message_from_database.message.auth.provider_public_key_bytes,
+        requestor_public_key = client_public_key,
+        status               = ReceiveStatus,
+        delivered            = True,
     )
     force_get_task_result_failed.sig = None
     return force_get_task_result_failed
 
 
-def handle_receive_force_get_task_result_upload_for_requestor(decoded_message: message.concents.ForceGetTaskResultUpload) -> message.concents.ForceGetTaskResultUpload:
+def handle_receive_force_get_task_result_upload_for_requestor(
+    request,
+    decoded_message:                message.concents.ForceGetTaskResultUpload,
+    previous_message_from_database: Message
```

Comment:
It's not a `Message`. It's message status.

Reply:
:+1: